app/models/manageiq/providers/cloud_manager/virtual_template.rb

```diff
@@ -0,0 +1,14 @@
+class ManageIQ::Providers::CloudManager::VirtualTemplate < ::MiqTemplate
+  validate :single_template, :on => :create
+  default_value_for :cloud, true
+
+  TYPES = {
+    :amazon => 'ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate'
```

Comment:
This breaks the pluggable provider concept.  This might be ok for now to get a first pass of this merged, but has to be rethought in the final version.  cc @durandom